mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -890,7 +890,7 @@ object LogisticRegression extends DefaultParamsReadable[LogisticRegression] {
   override def load(path: String): LogisticRegression = super.load(path)
 
   private[classification] val supportedFamilyNames =
-    Array("auto", "binomial", "multinomial").map(_.toLowerCase(Locale.ROOT))
+    Array("auto", "binomial", "multinomial")
```

Comment:
I am not sure about this. If we should keep `toLowerCase` here, we may also do this in `GeneralizedLinearRegression` and others